File: frontend/aviato/src/app/model/flight-dto.ts
import {RouteDto} from "./RouteDto";

export interface FlightDto {
  flightId: number,
  routeDto: RouteDto,
  airplaneModelName: string,
  scheduledDepartureTime: Date,
  scheduledArrivalTime: Date,
  actualArrivalTime: Date
  actualDepartureTime: Date
}

File: backend/src/main/java/ru/yakimov/aviato/dto/FlightDto.java
package ru.yakimov.aviato.dto;

import lombok.Builder;
import lombok.Value;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Embedded;

import java.util.Date;

;

@Value
@Builder(toBuilder = true)
public class FlightDto {
    @Id
    Integer flightId;
    @Embedded(prefix = "r_", onEmpty = Embedded.OnEmpty.USE_NULL)
    RouteDto routeDto;
    String airplaneModelName;
    Date scheduledDepartureTime;
    Date scheduledArrivalTime;
}
